src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -129,6 +152,7 @@ public void Dispose()
             StopIdleTimer();
 
             _disposed = true;
+            _codeDocumentTaskCompletionSource?.SetResult(CodeDocument);
```

Comment:
This should take the lock since `_disposed` and `_codeDocumentTaskCompletionSource` are now tied together